test/src/main/java/org/apache/accumulo/test/functional/PerTableCryptoIT.java

```diff
@@ -0,0 +1,213 @@
+    TableId tableId1;
+    TableId tableId2;
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      // maybe with more splits WAL will be easier to observe?
+      props.put(Property.TABLE_SPLIT_THRESHOLD.getKey(), "75K");
+
+      // table 0 plain text
+      c.tableOperations().create(tables[0]);
+      TableId tableId0 = TableId.of(c.tableOperations().tableIdMap().get(tables[0]));
+      VerifyIngest.VerifyParams params = new VerifyIngest.VerifyParams(getClientProps(), tables[0]);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
+      // verify data was not encrypted
+      System.out.println("-------------" + checkTableEncryption(tableId0, false));
+
+      // encrypt table 1
+      props.clear();
+      props.put(TABLE_SERVICE_NAME_PROP, AESCryptoService.class.getName());
+
+      tableConfig.setProperties(props);
+      c.tableOperations().create(tables[1], tableConfig);
+      params = new VerifyIngest.VerifyParams(getClientProps(), tables[1]);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
+      // clone encrypted table to table 2
+      var cloneConfig = CloneConfiguration.builder().setFlush(true).build();
+      c.tableOperations().clone(tables[1], tables[2], cloneConfig);
+
+      c.tableOperations().compact(tables[2], new CompactionConfig());
+      params = new VerifyIngest.VerifyParams(getClientProps(), tables[2]);
+      VerifyIngest.verifyIngest(c, params);
+
+      tableId1 = TableId.of(c.tableOperations().tableIdMap().get(tables[1]));
+      tableId2 = TableId.of(c.tableOperations().tableIdMap().get(tables[2]));
+    }
+
+    // verify files were encrypted in tables 1 and 2
+    System.out.println("-------------" + checkTableEncryption(tableId1, true));
+    System.out.println("-------------" + checkTableEncryption(tableId2, true));
+  }
+
+  /**
+   * Test that the OfflineIterator works with crypto
+   */
+  @Test
+  public void testOfflineIterator() throws Exception {
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      Map<String,String> props = new HashMap<>();
+      NewTableConfiguration tableConfig = new NewTableConfiguration();
+      // tableConfig.createOffline();
+      props.put(TABLE_SERVICE_NAME_PROP, AESCryptoService.class.getName());
+
+      tableConfig.setProperties(props);
+      c.tableOperations().create(tableName, tableConfig);
+      var params = new VerifyIngest.VerifyParams(getClientProps(), tableName);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+      TableId tableId = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+      c.tableOperations().offline(tableName, true);
+
+      var oScanner = new OfflineScanner((ClientContext) c, tableId, Authorizations.EMPTY);
+      int count = 0;
+      for (var e : oScanner) {
+        count++;
+      }
```

Comment:
```suggestion
      long count = oScanner.stream().count();
```
Thought this was a nice use of Scanners stream